### PR TITLE
Add unit test to check updates of decentralizationLevel

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

#1693 can be moved to "QA" column once this is merged.

### Overview

- Enhances the "unit test" to check that protocol parameters can be updated to include decentralization level.
- Removes "todo" comment from `NETWORK_PARAMS - Able to fetch network parameters` integration test case. The `decentralisationParam` in the shelley integration test genesis is 1.0 ⇒ 0% decentralised.

### CODE REVIEW

File: lib/byron/test/unit/Cardano/Wallet/Byron/NetworkSpec.hs [hunk 1]
```diff
@@ -80,3 +82,8 @@ zeroTxParameters :: TxParameters
 zeroTxParameters = TxParameters
     (LinearFee (Quantity 0) (Quantity 0) (Quantity 0))
     (Quantity 0)
+
+-- | A value that is not the same as what's in the test data genesis.
+halfD :: DecentralizationLevel
+halfD = DecentralizationLevel p
+    where Right p = mkPercentage (1/2)
```
Comment (KtorZ) [inline lib/byron/test/unit/Cardano/Wallet/Byron/NetworkSpec.hs]: It is sort of weird to have this unit test in the `byron` package; the decentralization level is something that exists only in Shelley.

File: lib/shelley/test/data/cardano-node-shelley/genesis.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ protocolParams:
   protocolVersion:
     minor: 0
     major: 0
-  decentralisationParam: 1
+  decentralisationParam: 1 # means 0% decentralised
```
Comment (KtorZ) [inline lib/shelley/test/data/cardano-node-shelley/genesis.yaml]: Weird that github shows this diff, whereas master `decentralisationParam` is actually `0.25` :thinking:

File: lib/byron/test/unit/Cardano/Wallet/Byron/NetworkSpec.hs (hunk 1 above)
Comment (rvl) [reply lib/byron/test/unit/Cardano/Wallet/Byron/NetworkSpec.hs]: OK I moved it to the shelley package.

File: lib/shelley/test/data/cardano-node-shelley/genesis.yaml (hunk 2 above)
Comment (rvl) [reply lib/shelley/test/data/cardano-node-shelley/genesis.yaml]: Well github doesn't show merge conflicts. It can only show a diff to the merge base.

**Review by KtorZ — APPROVED** [review]
bors r+

**Comment by KtorZ** [comment]
```
  test/unit/Cardano/Wallet/Shelley/NetworkSpec.hs:50:5: 
  1) Cardano.Wallet.Shelley.Network.getTxParameters Correct values are queried
       uncaught exception: ProcessHasExited
       ProcessDidNotStart "cardano-node" cardano-node: createProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
```

Ho. We need to provision unit tests with `cardano-node` as well.

**Pull request merged**